### PR TITLE
Message Passing Interface for ROOT

### PR DESCRIPTION
**Pull request by omazapa**
Hi Guys,

This is a pull request for ROOTMpi.

Best Omar.

### CODE REVIEW

File: mpi/rmpi/inc/Mpi/Globals.h [hunk 1]
```diff
@@ -0,0 +1,268 @@
...
+
+template <class T>
+MPI_Datatype GetDataType()
+{
+   if (typeid(T) == typeid(int) || typeid(T) == typeid(Int_t)) return MPI_INT;
+   if (typeid(T) == typeid(float) || typeid(T) == typeid(Float_t)) return MPI_FLOAT;
+   if (typeid(T) == typeid(double) || typeid(T) == typeid(Double_t)) return MPI_DOUBLE;
+   if (typeid(T) == typeid(bool) || typeid(T) == typeid(Bool_t)) return MPI_BYTE;
+   if (typeid(T) == typeid(char) || typeid(T) == typeid(Char_t)) return MPI_CHAR;
+   if (typeid(T) == typeid(short) || typeid(T) == typeid(Short_t)) return MPI_SHORT;
```
Comment (bluehood) [inline mpi/rmpi/inc/Mpi/Globals.h]: Hi, couldn't you just specialize that template for the different types? I don't think using `typeid` on a template type is necessary.

**Comment by Axel-Naumann** [comment]
I'd like to suggest that we first have a discussion on the feature before diving into the code review. (And Github isn't necessarily the best place for that discussion.)

**Comment by bluehood** [comment]
@Axel-Naumann sure sorry, I just took a peek at the code and commented on a thing I saw, didn't mean it as anything official :sweat_smile:

**Comment by omazapa** [comment]
Hi all,
I will to prepare a better PR, 
and I will to close this to reduce the noise in the emails.

Best,
Omar.

**Pull request closed without merging**